scripts/src/ui/page/timed.multichoice.question.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import { FullPageMultiChoiceProps, MultichoiceQuestion } from "./multichoice.question";
+
+import "../../../../styles/src/page/timed.multichoice.question.styl";
+
+const quizTimerTime = 300;
+const quizTimerWarnAt = 5;
+
+interface TimerProps {
+	time: number;
+
+	callback(): void;
+}
+
+class Timer extends React.Component<TimerProps, any> {
+	constructor(props: TimerProps) {
+		super(props);
+		this.state = {
+			started: false,
+			time: this.props.time,
+		};
+	}
+
+	public componentDidUpdate() {
+		if (!this.state.started) {
+			setTimeout(this.updateTime.bind(this), 1000);
```

Comment:
it will be better to save the `.bind` reference in the ctor instead of re-creating it on every update:
```
constructor(props: TimerProps) {
	super(props);

	this.updateTime = this.updateTime.bind(this);
	...
}
```